app/controllers/comments_controller.rb

```diff
@@ -142,11 +152,37 @@ def check_permission_to_delete
     access_denied(redirect: @comment) unless logged_in_as_admin? || current_user_owns?(@comment) || current_user_owns?(@comment.ultimate_parent)
   end
 
-  # Comments cannot be edited after they've been replied to
+  # Comments cannot be edited after they've been replied to or if they are frozen.
   def check_permission_to_edit
-    return if @comment&.count_all_comments&.zero?
-    flash[:error] = ts("Comments with replies cannot be edited")
-    redirect_to(request.env["HTTP_REFERER"] || root_path)
+    if @comment&.iced?
+      flash[:error] = ts("Frozen comments cannot be edited.")
+      redirect_to(request.env["HTTP_REFERER"] || root_path)
+    elsif !@comment&.count_all_comments&.zero?
+      flash[:error] = ts("Comments with replies cannot be edited")
+      redirect_to(request.env["HTTP_REFERER"] || root_path)
+    end
+  end
+
+  # Comments on works can be frozen by admins with proper authorization or the
+  # work creator.
+  # Comments on tags can be frozen by admins with proper authorization.
+  # Comments on admin posts can be frozen by any admin.
+  def check_permission_to_freeze
+    return if permission_to_modify_frozen_status
+
+    flash[:error] = ts("Sorry, you don't have permission to freeze that comment thread.")
```

Comment:
I feel like it'd be cleaner to use i18n here, rather than having two separate functions that do almost the same thing apart from the text being displayed. Maybe just `flash[:error] = t(".not_permitted")` or some such?

(That would also mean you could move the contents of `permission_to_modify_frozen_status` back into here, so that the logic and the error message aren't so far apart.)

Reply:
It was more of a philosophical decision than one related to the message. I originally had them combined only to [change them when I split freeze and unfreeze into separate actions](https://github.com/otwcode/otwarchive/pull/3893/commits/d401a73f382b148e3e4ebda02bb26a5d8944baf3#diff-34e52388abd1ae0658edd56e12cd6d935fac98673361084ba07a99ca4bb836e7L168). 

If we think it's better to keep them together, though, I can change it back.